stream-chat-android-client/src/main/java/io/getstream/chat/android/client/notifications/LoadNotificationDataWorker.kt

```diff
@@ -0,0 +1,108 @@
+package io.getstream.chat.android.client.notifications
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingWorkPolicy
+import androidx.work.ForegroundInfo
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.R
+import io.getstream.chat.android.client.logger.ChatLogger
+
+internal class LoadNotificationDataWorker(
+    private val context: Context,
+    workerParams: WorkerParameters,
+) : CoroutineWorker(context, workerParams) {
+
+    private val logger = ChatLogger.get("LoadNotificationDataWorker")
+
+    private val channelId: String = inputData.getString(DATA_CHANNEL_ID)!!
```

Comment:
I think it would make this class simpler if there was no class variables and all those variables become local variables inside `doWork` and they are passed along to the methods so we have referential transparency and the methods are independent from class that contains then.